packages/shell-evaluator/src/shell-evaluator.ts

```diff
@@ -0,0 +1,133 @@
+import AsyncWriter from '@mongosh/async-rewriter';
+import Mapper from '@mongosh/mapper';
+import { signatures, Help } from '@mongosh/shell-api';
+
+class ShellEvaluator {
+  public mapper: any;
+  private asyncWriter: AsyncWriter;
+  private bus: any;
+  private container: any;
```

Comment:
Let's add a type there.

Reply:
Typescript has duck-typing everywhere and types are matched by shape, so interfaces does not have to be implemented explicitly and we can have an object implementing multiple interfaces.

That means that unlike java we can define interfaces for what we expect from dependencies we have as parameters without affecting the 
implementation.

For example if you expect `container` to be an object having `toggleTelemetry(boolean)` you can enforce that rule with an interface or type there:

```
interface Container {
   toggleTelemetry(boolean): void
}
```

The implementor of `Container` does not need to know about that at all, it means that in the caller we can pass any object of any class that has a `toggleTelemetry(boolean)` method and could have any other methods too.